R/three-legged-oauth.R

```diff
@@ -1,21 +1,35 @@
 
 #' Create a token for a user via the browser flow.
-#' @param scopes List of scopes required for the returned token.
-#' @param oauth_app `httr::oauth_app` to use for this token fetch. (Optional.)
+#' @param scopes A character vector of scopes to request.
+#' @param app An OAuth consumer application, created by [httr::oauth_app()].
+#' @param cache A logical value or a string. `TRUE` means to cache using the
+#'   default cache file, `.httr-oauth`. `FALSE` means don't cache, and `NA`
+#'   means to guess using some sensible heuristics. A string means use the
+#'   specified path as the cache file.
 #' @param ... Additional arguments (ignored)
 #' @export
-credentials_user_oauth2 <- function(scopes, oauth_app = NULL, ...) {
-
-  app <- httr::oauth_app(
-    "google",
-    key = "603366585132-nku3fbd298ma3925l12o2hq0cc1v8u11.apps.googleusercontent.com",
-    secret = "as_N12yfWLRL9RMz5nVpgCZt"
-  )
+#' @examples
+#' \dontrun{
+#' scope <- "https://www.googleapis.com/auth/drive"
+#' app <- httr::oauth_app(
+#'   appname = "my_awesome_app",
+#'   key = "keykeykeykeykeykey",
+#'   secret = "secretsecretsecret"
+#' )
+#'
+#' ## credentials_user_oauth2() is one of the functions token_fetch() will try
+#' token_fetch(scope, app = app)
+#' }
+credentials_user_oauth2 <- function(scopes,
+                                    app,
+                                    cache = getOption("httr_oauth_cache"),
+                                    ...) {
   token <- httr::oauth2.0_token(
     endpoint = httr::oauth_endpoints("google"),
     app = app,
     scope = scopes,
-    use_oob = FALSE
+    use_oob = FALSE,
```

Comment:
+1 on trying to pass this through to `httr`; I wasn't sure the cleanest way to do it.

In particular, passing `...` on to `oauth2.0_token` is nice, but had three issues:
* we can never add new args unless we know they don't collide with `oauth2.0_token` args
* bad times if we ever migrate off `oauth2.0_token`
* I didn't know how (or if it's possible to) get roxygen to inherit args from a function in another package for doc purposes.

I think a reasonable option here is probably to just have a `token_args` option, and anything in that list gets passed along to `oauth2.0_token`. This gives us a nice escape hatch in the event of `oauth2.0_token` being changed: we can deprecate that arg.

WDYT?